dd-sdk-android-navigation/src/main/kotlin/com/datadog/android/androidx/navigation/NavigationViewTrackingStrategy.kt

```diff
@@ -0,0 +1,112 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.androidx.navigation
+
+import android.app.Activity
+import android.os.Bundle
+import androidx.annotation.IdRes
+import androidx.navigation.ActivityNavigator
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.DialogFragmentNavigator
+import androidx.navigation.fragment.FragmentNavigator
+import com.datadog.android.rum.ActivityLifecycleTrackingStrategy
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.ViewTrackingStrategy
+import java.lang.IllegalStateException
+
+class NavigationViewTrackingStrategy(
+    @IdRes private val navigationViewId: Int
+) :
+    ActivityLifecycleTrackingStrategy(),
+    ViewTrackingStrategy,
+    NavController.OnDestinationChangedListener {
+
+    // region ActivityLifecycleTrackingStrategy
+
+    override fun onActivityStarted(activity: Activity) {
+        super.onActivityStarted(activity)
+        val navController = activity.findNavControllerOrNull(navigationViewId)
+        navController?.addOnDestinationChangedListener(this)
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        super.onActivityPaused(activity)
+        val navController = activity.findNavControllerOrNull(navigationViewId)
+        navController?.currentDestination?.let {
+            GlobalRum.get().stopView(it)
+        }
+    }
+
+    override fun onActivityStopped(activity: Activity) {
+        super.onActivityStopped(activity)
+        val navController = activity.findNavControllerOrNull(navigationViewId)
+        navController?.removeOnDestinationChangedListener(this)
+    }
+
+    // endregion
+
+    // region OnDestinationChangedListener
+
+    override fun `onDestinationChanged`(
+        controller: NavController,
+        destination: NavDestination,
+        arguments: Bundle?
+    ) {
+        controller.currentDestination?.let { GlobalRum.get().stopView(it) }
+
+        val attributes = arguments.asRumAttributes()
+        val name = destination.getRumViewName()
+
+        GlobalRum.get().startView(destination, name, attributes)
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun Activity.findNavControllerOrNull(@IdRes viewId: Int): NavController? {
+        return try {
+            findNavController(viewId)
+        } catch (e: IllegalArgumentException) {
+            null
+        } catch (e: IllegalStateException) {
+            null
+        }
+    }
+
+    private fun Bundle?.asRumAttributes(): Map<String, Any?> {
+        if (this == null) return emptyMap()
+
+        val attributes = mutableMapOf<String, Any?>()
+
+        keySet().forEach {
+            attributes["$ARGUMENT_TAG.$it"] = get(it)
```

Comment:
hmmm....I am not sure this is GDPR compliant. What if we have sensible information in the arguments ?

Reply:
Good point. 
I'll add an argument to the class to let the developper enable this feature, which I think is really usefull. 
Also most of the time the data passed in arguments are item ids, and not necessarily "sensitive" data .